py3dtilers/IfcTiler/IfcTiler.py

```diff
@@ -24,19 +30,21 @@ def __init__(self):
                                  type=str,
                                  help='targeted unit of the 3DTiles produced')
 
-    def from_ifc(self, path_to_file, originalUnit, targetedUnit):
+    def from_ifc(self, path_to_file, grouped_by, originalUnit, targetedUnit):
         """
         :param path: a path to a directory
 
         :return: a tileset.
         """
-
-        pre_tileset, centroid = IfcObjectsGeom.retrievObjByType(path_to_file, originalUnit, targetedUnit)
+        if(grouped_by == 'IfcTypeObject'):
+            pre_tileset, centroid = IfcObjectsGeom.retrievObjByType(path_to_file, originalUnit, targetedUnit)
+        elif(grouped_by == 'IfcGroup'):
+            pre_tileset, centroid = IfcObjectsGeom.retrievObjByGroup(path_to_file, originalUnit, targetedUnit)
 
         objects = [objs for objs in pre_tileset.values() if len(objs) > 0]
         objects_to_tile = IfcObjectsGeom(objects)
 
-        return self.create_tileset_from_geometries(objects_to_tile)
+        return self.create_tileset_from_geometries(objects_to_tile, centroid)
```

Comment:
Why is the centroid given as an argument of `create_tileset_from_geometries` ? There, it will be interpreted as `extension_name=centroid`